### ISSUE TITLE
Mobilepay payments are not handled correctly

### ISSUE THREAD

**Issue by panuoksala** [paytrail/paytrail-for-woocommerce#67]
**Describe the bug**
Paytrail plugin reports successfull Mobilepay payment as failed payment into Woocommerce. This leads into situation, that customer has paid the product, but it won't reduce Woocommerce stocks and shows order in unpaid state.

For example, order notes shows that this payment has failed, because time limit has been reached, but actually the payment has been successfully done. I can get payment verification from https://api.checkout.fi for this payment.

![image](https://user-images.githubusercontent.com/15687235/200849075-e4586b61-1b50-4fbf-91d6-bbade51cf469.png)

**Steps to Reproduce**

I don't have exact details what customers are doing, but I think they are keeping the Mobilepay window open or something like that. Is it possible that browser is closed and then the payment is verified from mobilephone? This would make sense that the payment is received, but Woocommerce does not know about it.

**Expected behaviour**

Woocommerce should receive valid payment info if the Mobilepay payment is successfully done.

**Actual behaviour**

Customer order is left at unpaid status, Woocommerce stock is not reduced which leads into situation that same product can be sold twice even though that stock count is 1.

**Comment by panuoksala** [comment]
Is it possible that customer closes the browser and finishes payment with mobile device?

**Comment by loueranta-paytrail** [comment]
@panuoksala You are correct, sometimes customers fail to finish the payment flow and close the browser before they are redirected back to the webshop via our service. In those cases the payment remains open in WooCommerce until we confirm it via callback call. 

The callback is sent as soon as identify successful payment from the bank statements, which might happen with a delay of couple of days, due to weekends and holidays.

Unfortunately there is no way to completely eliminate issues caused by this delay, but one thing you can do, is to raise the "Hold stock" setting in the WooCommerce settings. (WooCommerce -> Settings -> Products -> Inventory)
